base/samples/Transformation.hpp

```diff
@@ -0,0 +1,90 @@
+#ifndef BASE_SAMPLES_TRANSFORMATION_HPP
+#define BASE_SAMPLES_TRANSFORMATION_HPP
+
+#include <string>
+#include <base/Time.hpp>
+#include <base/TransformWithCovariance.hpp>
+#include <base/samples/RigidBodyState.hpp>
+
+namespace base { namespace samples {
+
+/**
+ * Represents a transformation in 3D of a local coordinate frame inside a global coordinate frame.
+ * A frame ID shall be globally unique and corresponds to a coordinate system.
+ * Following the [Rock's conventions](http://rock.opendfki.de/wiki/WikiStart/Standards)
+ * coordinate systems are right handed with X forward, Y left and Z up.
+ *
+ * The transformation \f$ ^{global}_{local}T \f$ in this structure transforms a vector \f$ v_{local} \f$
+ * from the local into the global frame:
+ * \f$ v_{global} = ^{global}_{local}T \cdot v_{local} \f$
+ * In other words the transformation is the pose of the local frame expressed in the global frame.
+ * Example of a transformation chain:
+ * \f$ ^{world}_{sensor}T = ^{world}_{body}T \cdot ^{body}_{sensor}T \f$
+ */
+class Transformation
+{
+public:
+    /** Reference timestamp of the transformation sample */
+    base::Time time;
+
+    /** ID of the global reference frame */
+    std::string global_frame_id;
```

Comment:
> Besides the nomenclature, another intention of Transformation is to have a type which represents the full covariance (with cross-covariances) of 3D transformations while not having to store velocity information (which often is not needed/not available).

History: the reason why RBS exists as a pose+velocity is that *if* you have a pose+velocity estimator using a RBS, that estimator will not* be able to be connected to a consumer of a Transformation. Since having a pose+velocity estimate is not common but not rare either, we packed them in the same type.

RBS represents a frame change operation, which changes poses (or subparts of it) expressed in the source frame into the equivalent expressed in the target frame.

Which means that the pose part of it is the pose of the source in the target

The logical extension of this (since a RBS is the pose of A in B) is that the velocity part should be the velocity of A in B as well.